contracts/src/river/registry/facets/stream/StreamRegistry.sol

```diff
@@ -127,8 +126,8 @@ contract StreamRegistry is IStreamRegistry, RegistryModifiers {
       stream.flags |= StreamFlags.SEALED;
     }
 
-    // Delete genesis miniblock bytes if the stream is moving beyond genesis
-    if (lastMiniblockNum == 1) {
+    // Delete genesis miniblock bytes if the stream is moving beyond genesis.
+    if (genesisStored) {
```

Comment:
```suggestion
    // Delete genesis miniblock if `stream` still contains the genesis block after `stream` has advanced since genesis.
    if (stream.lastMiniblockNum == 0) {
```
Inlining for better clarity and perf.